### PR TITLE
Feature/feature register user

### PR DESCRIPTION
**Pull request by santidrj**

### CODE REVIEW

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ src/main/java/org/pesmypetcare/webservice/controller/usermanager/UserRestController.java  100
         

Complexity increasing per file
==============================
- src/main/java/org/pesmypetcare/webservice/service/UserServiceImpl.java  1
- src/main/java/org/pesmypetcare/webservice/dao/UserDaoImpl.java  1
- src/main/java/org/pesmypetcare/webservice/error/RestExceptionHandler.java  1
- src/main/java/org/pesmypetcare/webservice/error/FirebaseExceptionHandler.java  2
- src/main/java/org/pesmypetcare/webservice/securingservice/FirebaseFactory.java  2
- src/main/java/org/pesmypetcare/webservice/controller/appmanager/MyPetCareRestController.java  1
- src/test/java/org/pesmypetcare/webservice/controller/appmanager/MyPetCareRestControllerTest.java  2
- src/main/java/org/pesmypetcare/webservice/error/ErrorBody.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5213347&bid=17094557)

**Review by xavier-campos-diaz — APPROVED** [review]

**Review by enrichp — APPROVED** [review]
;-)

**Review by AlbertPG — APPROVED** [review]
For me any problems

**Review by marclll — COMMENTED** [review]
Well Done!

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Coverage increased per file
===========================
+ src/main/java/org/pesmypetcare/webservice/controller/usermanager/UserRestController.java  100
         

Complexity increasing per file
==============================
- src/main/java/org/pesmypetcare/webservice/securingservice/FirebaseFactory.java  2
- src/main/java/org/pesmypetcare/webservice/controller/appmanager/MyPetCareRestController.java  1
- src/test/java/org/pesmypetcare/webservice/controller/appmanager/MyPetCareRestControllerTest.java  3
- src/main/java/org/pesmypetcare/webservice/error/RestExceptionHandler.java  1
- src/main/java/org/pesmypetcare/webservice/service/UserServiceImpl.java  1
- src/main/java/org/pesmypetcare/webservice/error/FirebaseExceptionHandler.java  2
- src/main/java/org/pesmypetcare/webservice/error/ErrorBody.java  1
- src/main/java/org/pesmypetcare/webservice/dao/UserDaoImpl.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5213347&bid=17094557)

File: src/test/java/org/pesmypetcare/webservice/controller/appmanager/MyPetCareRestControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+            if (deleteUser1) {
+                deleteUser("user1");
+                deleteUser1 = false;
+            }
+        } catch (FirebaseAuthException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @BeforeAll
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/controller/appmanager/MyPetCareRestControllerTest.java]: Codacy Issue found: [Method length is 16 lines (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5213347)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ src/main/java/org/pesmypetcare/webservice/controller/usermanager/UserRestController.java  100
         

Complexity increasing per file
==============================
- src/main/java/org/pesmypetcare/webservice/error/RestExceptionHandler.java  1
- src/main/java/org/pesmypetcare/webservice/error/FirebaseExceptionHandler.java  2
- src/main/java/org/pesmypetcare/webservice/service/UserServiceImpl.java  1
- src/test/java/org/pesmypetcare/webservice/controller/appmanager/MyPetCareRestControllerTest.java  3
- src/main/java/org/pesmypetcare/webservice/error/ErrorBody.java  1
- src/main/java/org/pesmypetcare/webservice/securingservice/FirebaseFactory.java  2
- src/main/java/org/pesmypetcare/webservice/controller/appmanager/MyPetCareRestController.java  1
- src/main/java/org/pesmypetcare/webservice/dao/UserDaoImpl.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5213347&bid=17094557)

**Pull request merged**